include/zephyr/bluetooth/audio/audio.h

```diff
@@ -371,20 +350,37 @@ struct bt_audio_codec_cfg {
 	uint16_t cid;
 	/** Codec Company Vendor ID */
 	uint16_t vid;
-#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT)
-	/** Codec Specific Configuration Data count */
-	size_t   data_count;
-	/** Codec Specific Configuration Data */
-	struct bt_audio_codec_data data[CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT];
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_COUNT */
-#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT)
-	/** Codec Specific Configuration Metadata count */
-	size_t   meta_count;
-	/** Codec Specific Configuration Metadata */
-	struct bt_audio_codec_data meta[CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT];
-#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_COUNT */
+#if CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE > 0
+	/** Codec Specific Capabilities Data count */
+	size_t data_len;
+	/** Codec Specific Capabilities Data */
+	uint8_t data[CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE];
+#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_DATA_SIZE > 0 */
+#if defined(CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE)
+	/** Codec Specific Capabilities Metadata count */
+	size_t meta_len;
+	/** Codec Specific Capabilities Metadata */
+	uint8_t meta[CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE];
+#endif /* CONFIG_BT_AUDIO_CODEC_CFG_MAX_METADATA_SIZE */
 };
 
+/**
+ * @brief Helper for parsing length-type-value data.
+ *
+ * @param ltv       Length-type-value (LTV) encoded data.
+ * @param size      Size of the @p ltv data.
+ * @param func      Callback function which will be called for each element
+ *                  that's found in the data. The callback should return
+ *                  true to continue parsing, or false to stop parsing.
+ * @param user_data User data to be passed to the callback.
+ *
+ * @retval 0 if all entries were parsed.
+ * @retval -EINVAL if the data is incorrectly encoded
+ * @retval -ECANCELED if parsing was prematurely cancelled by the callback
+ */
+int bt_audio_data_parse(const uint8_t ltv[], size_t size,
+			bool (*func)(struct bt_data *data, void *user_data), void *user_data);
```

Comment:
We already have `bt_data_parse` declared in `zephyr/bluetooth/bluetooth.h` and defined in `subsys/bluetooth/host/data.c`.
As the function you add, does not seem to be LeAudio specific, but rather flatbuf version of bt_audio_data_parse, thus I think it would be better to place it in the same place `bt_data_parse` is.

Reply:
See my response in https://github.com/zephyrproject-rtos/zephyr/pull/58331#discussion_r1302890151

I've already talked to @alwa-nordic before about supporting `const` data in `net_buf_simple`, but I think that is better done as a separate PR dedicated to bringing an API for that, rather than it being a part of this. I suggest to keep this (perhaps temporary API) for LE Audio for now.